src/utility/Sd2Card.cpp

```diff
@@ -631,16 +632,17 @@ uint8_t Sd2Card::writeBlock(uint32_t blockNumber, const uint8_t* src) {
   if (!writeData(DATA_START_BLOCK, src)) {
     goto fail;
   }
-
-  // wait for flash programming to complete
-  if (!waitNotBusy(SD_WRITE_TIMEOUT)) {
-    error(SD_CARD_ERROR_WRITE_TIMEOUT);
-    goto fail;
-  }
-  // response is r2 so get and check two bytes for nonzero
-  if (cardCommand(CMD13, 0) || spiRec()) {
-    error(SD_CARD_ERROR_WRITE_PROGRAMMING);
-    goto fail;
+  if (sync) {

```

Comment:
@cmaglie @facchinm this part is a bit dangerous, since we no longer wait for the result of the write command.

In the SD card I used for testing, the `waitNotBusy` would spin for 10 - 20ms.

Reply:
We should take extra care and understand how people really use this library (like continuous blocking writes or other patterns).
Said that, I love this patch!

Reply:
I think we can only activate this non-blocking behaviour if someone calls `availableForWrite()`. This will allow older sketches to continue with the existing code path.

Stay tuned, I'll push some more changes later today.